UnitsNet.Tests/CustomCode/AmountOfSubstanceTests.cs

```diff
@@ -70,45 +60,63 @@ public void NumberOfParticlesInTwoMolesIsDoubleAvogadroConstant()
             Assert.Equal(AmountOfSubstance.AvogadroConstant * 2, numberOfParticles);
         }
 
-        [Fact]
-        public void TenMolesOfOxygenToMassEqualToExpected()
+        [Theory]
+        [InlineData(10, AmountOfSubstanceUnit.Mole,
+                    KnownQuantities.MolarMassOfOxygen, MolarMassUnit.GramPerMole,
+                    10 * KnownQuantities.MolarMassOfOxygen, MassUnit.Gram)]     // 10 Moles of Oxygen weight 10 times as much as 1 Mole of Oxygen (MolarMass)
+        public void MassFromAmountOfSubstanceAndMolarMass(
+            double amountOfSubstanceValue, AmountOfSubstanceUnit amountOfSubstanceUnit,
+            double molarMassValue, MolarMassUnit molarMassUnit,
+            double expectedMass, MassUnit expectedMassUnit, double tolerence = 1e-5)
         {
-            AmountOfSubstance tenMoles = AmountOfSubstance.FromMoles(10);
-            MolarMass molarMass = MolarMass.FromGramsPerMole(MolarMassOfOxygen);
-            AssertEx.EqualTolerance(10 * MolarMassOfOxygen, (tenMoles * molarMass).Grams, MolesTolerance);
-        }
+            AmountOfSubstance amountOfSubstance = new AmountOfSubstance(amountOfSubstanceValue, amountOfSubstanceUnit);
+            MolarMass molarMass = new MolarMass(molarMassValue, molarMassUnit);
+            
+            Mass mass = amountOfSubstance * molarMass;
 
-        [Fact]
-        public void TenGramsOfOxygenToMolesEqualToExpected()
-        {
-            Mass tenGrams = Mass.FromGrams(10);
-            MolarMass molarMass = MolarMass.FromGramsPerMole(MolarMassOfOxygen);
-            AssertEx.EqualTolerance(MolesInTenGramsOfOxygen, (tenGrams / molarMass).Moles, MolesTolerance);
+            AssertEx.EqualTolerance(expectedMass, mass.As(expectedMassUnit), tolerence);
         }
 
-        [Fact]
-        public void HClSolutionVolumeIsEqualToExpected()
+        [Theory]
+        [InlineData(5, MassUnit.Gram,
```

Comment:
To be clear, this is fine, but I didn't mean that `[InlineData]` was my preferred method of testing. Only that the values would be inlined/inserted into the test code itself, so that you can read all the numbers that go into the calculation for each test method rather than reusing fields. `[Fact]` is probably more "correct" than `[Theory]` + `[InlineData]` if there is only one test case. If you have multiple test cases/inputs, then `[InlineData]` is a good fit.

At any rate, this works and I'm not going to bother you to change it again.